src/sentry/monitors/tasks.py

```diff
@@ -390,3 +407,51 @@ def mark_checkin_timeout(checkin_id: int, ts: datetime, **kwargs):
         )
 
         mark_failed(checkin, ts=most_recent_expected_ts)
+
+
+@instrumented_task(
+    name="sentry.monitors.tasks.detect_broken_monitor_envs",
+    max_retries=0,
+    time_limit=15 * 60,
+    soft_time_limit=10 * 60,
+    record_timing=True,
+)
+def detect_broken_monitor_envs():
+    current_time = django_timezone.now()
+    open_incidents = MonitorIncident.objects.select_related("monitor").filter(
+        resolving_checkin=None,
+        starting_timestamp__lte=(current_time - timedelta(days=NUM_DAYS_BROKEN_PERIOD)),
+    )
```

Comment:
You'll need to add an index here, like 
```
            models.Index(
                fields=["starting_timestamp"], name="active_incident_idx", condition=Q(resolving_checkin__isnull=True)
            ),
```

Reply:
Does the condition on the index help with size of the index?

Reply:
Partially yeah. It's also pointless to have a compound index on two timestamp columns, because they're too high cardinality for the second column to make sense. So better to have the very specific index, and then we're also less likely to have the query planner go crazy and choose the wrong thing (very obvious for it to use this index when these clauses match so closely to the query)

Reply:
https://github.com/getsentry/sentry/pull/66444